src/classes/BDI_Donations.cls

```diff
@@ -236,22 +240,46 @@ public class BDI_Donations {
 
         // add any new payments to our map
         trackPayments(dataImportDonations);
-        
+
         // when we mark Payments paid, ensure they will close our Opp.
         npe01__Contacts_And_Orgs_Settings__c cs = UTIL_CustomSettingsFacade.getContactsSettings();
         if (cs.Payments_Auto_Close_Stage_Name__c == null) {
             cs.Payments_Auto_Close_Stage_Name__c = defaultClosedWonStage;
         }
-        
+
         // update payments with any data from the DI record
         updatePayments(dataImportDonations);
     }
 
+    /*******************************************************************************************************
+    * @description Determine if the data import batch number should be set on the opportunity or payment
+    * @param targetObject The opportunity or payment SObject record
+    * @param dataImport The data import record being processed
+    * @return Boolean TRUE when the Data Import Batch has a batch number, is not a dry run,
+    * and the current SObject is an Opportunity or Opportunity Payment
+    */
+    private Boolean shouldSetDonationBatchNumber(SObject targetObject, DataImport__c dataImport) {
+        SObjectType objectType = targetObject.getSObjectType();
+
+        if (bdi.isDryRun ||
```

Comment:
Can we move this check ahead of the SObjectType assignment right above this (currently line 262)? Since this is inside the processing loop I think we should return as quickly as possible if we don't need to handle the batch number field

Reply:
sure thing. fixed!